registrations/signals.py

```diff
@@ -90,3 +91,25 @@ def calculate_registration_price_group_vat_prices(
     **kwargs: dict,
 ) -> None:
     instance.calculate_vat_and_price_without_vat()
+
+
+@receiver(
+    post_save,
+    sender=Registration,
+    dispatch_uid="create_signup_link_for_event",
+)
+def create_signup_link_for_event(
+    sender: type[Registration], instance: Registration, **kwargs: dict
+) -> None:
+    if not kwargs.get("created"):
+        # Only create the signup link if a new registration has been created.
+        return
+
+    instance.event.offers.filter(
+        Q(info_url_fi=None, info_url_sv=None, info_url_en=None)
+        | Q(info_url_fi="", info_url_sv="", info_url_en="")
+    ).update(
+        info_url_fi=get_signup_create_url(instance, "fi"),
+        info_url_sv=get_signup_create_url(instance, "sv"),
+        info_url_en=get_signup_create_url(instance, "en"),
+    )
```

Comment:
Should offer info url be updated also if e.g. info_url_fi="" and info_url_sv=None? This can be achieved for example with:
```suggestion
    instance.event.offers.filter(
        (Q(info_url_fi=None) | Q(info_url_fi=""))
        & (Q(info_url_sv=None) | Q(info_url_sv=""))
        & (Q(info_url_en=None) | Q(info_url_en=""))
    ).update(
        info_url_fi=get_signup_create_url(instance, "fi"),
        info_url_sv=get_signup_create_url(instance, "sv"),
        info_url_en=get_signup_create_url(instance, "en"),
    )
```

Reply:
Very nice catch -> improved as suggested